app/helpers/estimate_complexity.py

```diff
@@ -0,0 +1,187 @@
+# standard library imports
+import json
+from typing import List, Tuple
+import logging
+
+# external library imports
+import scipy
+import numpy as np
+
+# internal imports
+from app.helpers import complexity_models
+import app.settings as settings
+
+def get_standard_error(
+    outputs_of_model: np.array, actual_runtimes: List[float]
+) -> List[float]:
+    diff = outputs_of_model - actual_runtimes
+    return np.std(diff)/np.sqrt(len(diff))
+
+def min_max_normalise(data: List[float]) -> List[float]:
+    data = np.array(data)
+    min_nax_norm = (data-data.min()) / (data.max()-data.min()+0.00001)
+    return min_nax_norm.tolist()
+
+def prepare_data_for_complexity_estimation(
+    input_and_time_list: List[Tuple[str, str]],
+    input_type: str,
+) -> Tuple[List[int], List[float]]:
+    if input_type == settings.STRING_INPUT_CODE:
+        x_data = [len(i[0]) for i in input_and_time_list]
+    else:
+        x_data = [i[0] for i in input_and_time_list]
+    runtime_list = [i[1] for i in input_and_time_list]
+    normalised_runtime_list = min_max_normalise(runtime_list)
+    return x_data, normalised_runtime_list
+
+def get_complexity_estimates(
+    input_and_time_list: List[Tuple[str, str]],
+    input_type: str,
+) -> json:
+    x_data, runtime_list = prepare_data_for_complexity_estimation(
+        input_and_time_list, input_type
+    )
+    # get the arguments for each model that fit the curve best
+    #constant model
+    try:
+        constant_args, _ = scipy.optimize.curve_fit(
+            f=complexity_models.constant_model, method="lm", xdata=x_data,
+            ydata=runtime_list, p0=[0]
+        )
+    except RuntimeError as e:
+        if 'Optimal parameters not found' in e.args[0]:
```

Comment:
Note: add a comment to justify why this was done

Idea: Maybe make a catch-optimal-param-exception decorator to replace the try/except block from all the model fitting code?

Reply:
Since all of these try-except statements are in the same method, the decorator would be specific to this method and hence wouldn't be a good idea.
Skipping this.